http/server.lua

```diff
@@ -589,7 +589,53 @@ response_mt = {
     }
 }
 
-local function handler(self, request)
+local function is_function(obj)
+    return type(obj) == 'function'
+end
+
+local function get_request_logger(server_opts, route_opts)
+    local root_logger = log.debug
+    if server_opts.log_requests then
+        root_logger = log.info
+
+        if is_function(server_opts.log_requests) then
+            root_logger = server_opts.log_requests
+        end
+    end
+
+    if route_opts and route_opts.endpoint.log_requests ~= nil then
+        if is_function(route_opts.endpoint.log_requests) then
+            return route_opts.endpoint.log_requests
+        elseif route_opts.endpoint.log_requests == false then
+            return log.debug
+        end
+    end
+
+    return root_logger
+end
+
+local function get_error_logger(server_opts, route_opts)
+    local root_logger = log.debug
```

Comment:
Simplification suggestion:
1) check route first, if found - return immediately, then
2) check server, if found - return immediately, then
3) return default

No need for `root_logger` variable.